### PR TITLE
Back-merging release into develop

### PR DESCRIPTION
**Pull request by TobiasPalludan**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -3,20 +3,21 @@ import 'package:weekplanner/blocs/auth_bloc.dart';
...
 
   if (_isInDebugMode) {
-    // If in DEBUG mode
+    /// If in DEBUG mode
     environment.setFile('assets/environments.json').whenComplete(() {
       _runApp();
     });
   } else {
-    // Else Production
+    /// Else Production
```
Comment (boginw) [inline lib/main.dart]: Should not be block comment

File: lib/main.dart [hunk 2]
```diff
@@ -3,20 +3,21 @@ import 'package:weekplanner/blocs/auth_bloc.dart';
...
 import 'package:weekplanner/screens/login_screen.dart';
 
 void main() {
-  // Register all dependencies for injector
+  /// Register all dependencies for injector
   Bootstrap.register();
 
   if (_isInDebugMode) {
-    // If in DEBUG mode
+    /// If in DEBUG mode
```
Comment (boginw) [inline lib/main.dart]: Should not be block comment

File: lib/main.dart [hunk 3]
```diff
@@ -3,20 +3,21 @@ import 'package:weekplanner/blocs/auth_bloc.dart';
 import 'package:weekplanner/bootstrap.dart';
 import 'package:weekplanner/di.dart';
 import 'package:weekplanner/providers/environment_provider.dart' as environment;
+import 'package:weekplanner/routes.dart';
 import 'package:weekplanner/screens/choose_citizen_screen.dart';
 import 'package:weekplanner/screens/login_screen.dart';
 
 void main() {
-  // Register all dependencies for injector
+  /// Register all dependencies for injector
```
Comment (boginw) [inline lib/main.dart]: Should not be block comment

File: lib/main.dart [hunk 4]
```diff
@@ -29,9 +30,16 @@ void _runApp() {
       home: StreamBuilder<bool>(
           initialData: false,
           stream: di.getDependency<AuthBloc>().loggedIn,
-          builder: (_, AsyncSnapshot<bool> snapshot) =>
-              // In case logged in show ChooseCitizenScreen, otherwise login
-              snapshot.data ? ChooseCitizenScreen() : LoginScreen())));
+          builder: (BuildContext context, AsyncSnapshot<bool> snapshot) {
+            if (snapshot.data) {
+              /// In case logged in show ChooseCitizenScreen
```
Comment (boginw) [inline lib/main.dart]: Should not be block comment

File: lib/main.dart [hunk 5]
```diff
@@ -29,9 +30,16 @@ void _runApp() {
...
+              return ChooseCitizenScreen();
+            } else {
+              /// Not loggedIn pop context to login screen.
```
Comment (boginw) [inline lib/main.dart]: Should not be block comment

File: lib/blocs/weekplans_bloc.dart [hunk 6]
```diff
@@ -50,15 +50,15 @@ class WeekplansBloc extends BlocBase {
...
   /// The result are published in [_weekModel].
   void getAllWeekInfo(List<WeekNameModel> weekNameModels) {
-    if (weekNameModels == null) {
-      print(weekNameModels);
-      return;
-    }
     final List<WeekModel> weekModels = <WeekModel>[];
-    
+
     /// This is used by weekplan_selector_screen for adding a new weekplan.
```
Comment (boginw) [inline lib/blocs/weekplans_bloc.dart]: Should not be block comment

File: lib/blocs/weekplans_bloc.dart [hunk 7]
```diff
@@ -50,15 +50,15 @@ class WeekplansBloc extends BlocBase {
...
     if (_addWeekplan) {
-      weekModels.add(WeekModel(name: 'Tilføj Ugeplan'));
+      weekModels.add(WeekModel(name: 'Tilføj ugeplan'));
+      _weekModel.add(weekModels);
+    }
+    if (weekNameModels == null) {
```
Comment (boginw) [inline lib/blocs/weekplans_bloc.dart:60]: this terminator should come before the previous selection statement

**Review by Tekrus — DISMISSED** [review]

**Review by boginw — CHANGES_REQUESTED** [review]
I'm not going to go too thorough trough it, but there are some code-smells some places. Here are a few minor corrections

**Comment by Tekrus** [comment]
@boginw can you make a PR fixing those small things, so that we can get it merged into develop before the sprint introduction?

File: lib/blocs/weekplans_bloc.dart (hunk 7 above)
Comment (LukasRH) [reply lib/blocs/weekplans_bloc.dart:60]: This was how it originally was, however, this coursed the effect that if a user didn't have any weekplans, the button to create a new weekplan would not be made. Moving it down made sure that the button would always be made.

File: lib/blocs/weekplans_bloc.dart (hunk 6 above)
Comment (TobiasPalludan) [inline lib/blocs/weekplans_bloc.dart]:
```suggestion
    // This is used by weekplan_selector_screen for adding a new weekplan.
```

File: lib/main.dart (hunk 3 above)
Comment (TobiasPalludan) [inline lib/main.dart]:
```suggestion
  // Register all dependencies for injector
```

File: lib/main.dart (hunk 2 above)
Comment (TobiasPalludan) [inline lib/main.dart]:
```suggestion
    // If in DEBUG mode
```

File: lib/main.dart (hunk 1 above)
Comment (TobiasPalludan) [inline lib/main.dart]:
```suggestion
    // Else Production
```

File: lib/main.dart (hunk 4 above)
Comment (TobiasPalludan) [inline lib/main.dart]:
```suggestion
              // In case logged in show ChooseCitizenScreen
```

File: lib/main.dart (hunk 5 above)
Comment (TobiasPalludan) [inline lib/main.dart]:
```suggestion
              // Not loggedIn pop context to login screen.
```

**Review by Tekrus — APPROVED** [review]

**Review by boginw — APPROVED** [review]

**Pull request merged**